Makefile

```diff
@@ -276,10 +283,18 @@ lint: ## Run all the lint targets
 	$(MAKE) lint-shell
 
 GOLANGCI_LINT_FLAGS ?= --fast=true
+GOLANGCI_LINT_ABS_PATH := $(abspath $(GOLANGCI_LINT))
 .PHONY: lint-go
 lint-go: $(GOLANGCI_LINT)
 lint-go: ## Lint codebase
-	$(GOLANGCI_LINT) run -v $(GOLANGCI_LINT_FLAGS)
+	@for dir in $(GO_MOD_DIRS); do \
+		if [[ "$$dir" == ./external* || "$$dir" == ./hack/tools* ]]; then \
+			echo "Skipping $$dir"; \
+			continue; \
+		fi; \
+		echo "Running golangci-lint in $$dir"; \
+		(cd $$dir && $(GOLANGCI_LINT_ABS_PATH) run -v $(GOLANGCI_LINT_FLAGS)); \
```

Comment:
Are `GO_MOD_DIRS` relative paths?  If so I am curious how `cd $$dir` is working here since you pasted the output of running this target in your `Testing Done` section.

Reply:
Yes, they're relative paths as shown in the command output. I think it works because `cd $$dir` creates a subshell each time it gets called.